game-app/game-core/src/main/java/games/strategy/triplea/printgenerator/PlayerOrder.java

```diff
@@ -45,6 +46,7 @@ void saveToFile(final PrintGenerationData printData) throws IOException {
         playerSet.add(currentGamePlayer);
       }
     }
+    playerSet.sort(new PlayerOrderComparator(printData.getData()));
```

Comment:
This seems to be the one really important line. It looks like this sort should be fine. The `PlayerOrderComparator` code is wonky, hard to understand. I don't like how there is an acquisition of a read lock that previously was not there, though that *should* not create any problems.